projects/laji/src/app/shared-modules/select/select-subcategories/select-subcategories.component.ts

```diff
@@ -0,0 +1,569 @@
+  ngOnChanges() {
+    if (this.disabled) {
+      this.selectedOptions = {};
+      this.open = false;
+    }
+
+    if (this.filtersValues.every(element => element === undefined)) {
+      this.selectedOptions = {};
+      this.tmpSelectedOption = {};
+      this.selectedOptions = {};
+    }
+    this.initOptions(this.selectedOptions !== undefined && Object.keys(this.selectedOptions).length > 0 ? this.selectedOptions : this.buildSelectedOptions());
+    this.selectedOptions =  this.tmpSelectedOption.constructor === Object && Object.keys(this.tmpSelectedOption).length === 0 ?
+    this.selectedOptions : this.tmpSelectedOption;
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+  }
+
+  toggleValue(id: string, category: string) {
+    this.selectedOptions = Object.keys(this.tmpSelectedOption).length === 0 && this.tmpSelectedOption.constructor === Object ?
+    this.selectedOptions : this.tmpSelectedOption;
+    this.selectedIdx[category] = 0;
+    if (!this.selectedOptions[category]) {
+      this.add(id, category);
+      return;
+    }
+
+    if (this.selectedOptions[category].indexOf(id) === -1) {
+      this.add(id, category);
+    } else {
+      this.remove(id, category);
+    }
+  }
+
+  add(id: string, category: string) {
+    if (this.multiple) {
+      if (typeof this.selectedOptions !== 'object' || this.selectedOptions === null) {
+        this.selectedOptions = {};
+      }
+      this.selectedOptions[category] = [...this.selectedOptions[category] || [], id];
+      if (category === 'GLOBAL') {
+        this.subCategories.forEach(element => {
+          if (element !== 'GLOBAL' && this.selectedOptions[element]) {
+            this.selectedOptions[element] = this.selectedOptions[element].indexOf(id) === -1 ?
+            [...this.selectedOptions[element] || [], id] : [...this.selectedOptions[element]];
+          }
+        });
+      }
+    } else {
+      this.selectedOptions[category] = [id];
+      if (category === 'GLOBAL') {
+        this.subCategories.forEach(element => {
+          if (element !== 'GLOBAL') {
+            this.selectedOptions[element] = [id];
+          }
+        });
+      }
+    }
+
+    this.emitSelectedOptions(category);
+  }
+
+  remove(id: string, category: string) {
+    this.selectedOptions[category] = this.selectedOptions[category].filter(value => value !== id);
+    if (category === 'GLOBAL') {
+      this.subCategories.forEach(element => {
+        if (element !== 'GLOBAL') {
+          this.selectedOptions[element] = this.selectedOptions[element].filter(value => value !== id);
+        }
+      });
+    }
+
+    this.emitSelectedOptions(category);
+  }
+
+  toggle(event, el, typeCheckbox = 0) {
+    if (event.target.classList.contains('no-propagation')) {
+      return;
+    }
+    this.open = !this.open;
+    this.typeCheckbox = typeCheckbox;
+    if (this.open && this.useFilter) {
+      ObservableInterval(10).pipe(takeUntil(this.unsubscribe$), take(1))
```

Comment:
timer could be better since you only need this triggered once. Also when is this used I could not find a way to get the part after the first if in here to active?